src/pages/receipts/csvFileReceipt.js

```diff
@@ -144,161 +151,116 @@ const updateInTransitMapping = (shippedBoxes) => {
+}
 
-    else if (i[fieldToConceptIdMapping.collectionType] === fieldToConceptIdMapping.clinical && i[fieldToConceptIdMapping.healthcareProvider] === nameToKeyObj["hfHealth"] && 
-    (i[fieldToConceptIdMapping.collectionId].split(' ')[1] === '0005')) {
-      vialMappingsHolder.push('6 ml Vacutainer', 'ACD', 'Whole Blood', '6')
-    }
+/**
+ * Maps specimen id to material type based on last 4 digits, collection type & health care provider
+ * @param {object} filteredResult - Object containg essential information (health care provider, collection type, & more)
+ * @returns {array} Returns an array containing all the vial mapping
+*/ 
 
-    else if (i[fieldToConceptIdMapping.collectionType] === fieldToConceptIdMapping.clinical && i[fieldToConceptIdMapping.healthcareProvider] === nameToKeyObj["hfHealth"] && 
-    (i[fieldToConceptIdMapping.collectionId].split(' ')[1] === '0006')) {
-      vialMappingsHolder.push('10 ml Vacutainer', 'No Additive', 'Urine', '6')
-    }
+const getVialTypesMappings = (filteredResult) => {
+  let vialMappingsHolder = []
+  const tubeId = filteredResult[fieldToConceptIdMapping.collectionId].split(' ')[1]
+  const collectionType = filteredResult[fieldToConceptIdMapping.collectionType]
+  const healthCareProvider = filteredResult[fieldToConceptIdMapping.healthcareProvider]
 
-    else{}
+  if (collectionType === fieldToConceptIdMapping.research && (tubeId === '0001' || tubeId === '0002' )) {
+    vialMappingsHolder.push('8.5 mL Serum separator tube', 'SST', 'Serum', '8.5')
+  } else if (collectionType === fieldToConceptIdMapping.research && tubeId === '0003') {
+    vialMappingsHolder.push('10 ml Vacutainer', 'Lithium Heparin', 'Whole Bl', '10')
+  } else if (collectionType === fieldToConceptIdMapping.research && tubeId === '0004') {
+    vialMappingsHolder.push('10 ml Vacutainer', 'EDTA = K2', 'Whole Bl', '10')
+  } else if (collectionType === fieldToConceptIdMapping.research && tubeId === '0005') {
+    vialMappingsHolder.push('6 ml Vacutainer', 'ACD', 'Whole Bl', '6')
+  } else if (collectionType === fieldToConceptIdMapping.research && tubeId === '0006') {
+    vialMappingsHolder.push('10 ml Vacutainer', 'No Additive', 'Urine', '10')
+  } else if (collectionType === fieldToConceptIdMapping.research && tubeId === '0007') {
+    vialMappingsHolder.push('15ml Nalgene jar', 'Crest Alcohol Free', 'Saliva', '15')
+  } else if (collectionType === fieldToConceptIdMapping.clinical && (tubeId === '0001' || tubeId === '0002' || tubeId === '0011' || tubeId === '0012' || tubeId === '0021')) {
+    vialMappingsHolder.push('5 mL Serum separator tube', 'SST', 'Serum', '5')
+  } else if (collectionType === fieldToConceptIdMapping.clinical && (tubeId === '0003' || tubeId === '0013')) {
+    vialMappingsHolder.push('4 ml Vacutainer', 'Lithium Heparin', 'Whole BI', '4')
+  } else if (collectionType === fieldToConceptIdMapping.clinical && (tubeId === '0004' || tubeId === '0014') || tubeId === '0024') {
```

Comment:
Is `(tubeId === '0004' || tubeId === '0014') || tubeId === '0024'` intended, or should be `(tubeId === '0004' || tubeId === '0014' || tubeId === '0024')`?